src/NServiceBus.Persistence.DynamoDB.Tests/Serialization/MapperTests.cs

```diff
@@ -330,6 +330,28 @@ public class ClassWithSetOfMemoryStream
         public ImmutableHashSet<MemoryStream> ImmutableHashSetOfStreams { get; set; }
     }
 
+    [Test]
+    public void Should_roundtrip_empty_sets_streams()
+    {
+        var classWithListOfMemoryStream = new ClassWithEmptySetOfMemoryStream();
+
+        var attributes = Mapper.ToMap(classWithListOfMemoryStream);
+
+        var deserialized = Mapper.ToObject<ClassWithEmptySetOfMemoryStream>(attributes);
+
+        CollectionAssert.AreEquivalent(classWithListOfMemoryStream.HashSetOfMemoryStreams, deserialized.HashSetOfMemoryStreams);
+        CollectionAssert.AreEquivalent(classWithListOfMemoryStream.ImmutableHashSetOfStreams, deserialized.ImmutableHashSetOfStreams);
```

Comment:
I do [understand you removed the extra assertions](https://github.com/Particular/NServiceBus.Persistence.DynamoDB/commit/0e6ed17bd7d1eaab33f52e1a3b05c12901a401a8). I disagree with the removal. When I yesterday mentioned that I did introduce lots of assertions and there might be other ways we can make those tests more expressive (like with an approval test) I did not mean we should remove those.

The entire idea of these mapping tests is to capture the to and from "physical" layout mapping. It is not a detail where things are stored or explicitly chosen not to be stored. Those details matter and should be backed into a test. That was the intent of these assertions and when you look at the other tests in this file they all behave the same except now with this change even in this PR [we have some tests that follow the previously outlined safety net](https://github.com/Particular/NServiceBus.Persistence.DynamoDB/pull/601/files#diff-e8daee2d58ee4aeacd5fbb23d4af08130f7f6b640783d813d5ffa4aca20b2960R738-R743) vs some that don't (example here) which for me is :-1: 

When you look at the behavior that we are introducing we need to verify that empty NS, SS and BS fields are stored in L with IsLSet = true. We are also missing tests that express the "flip" of storage. Happy to add those.

Reply:
@DavidBoike here we go https://github.com/Particular/NServiceBus.Persistence.DynamoDB/pull/605

Reply:
Agreed and restored by merging #605